crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -34,6 +34,56 @@ def check
       }
     end
 
+    def check_and_enable_system_repos(options = {})
+      # FIXME: get rid of platform_exception when SES upgraded to SP2
+      platform_exception = options.fetch(:platform_exception, nil)
+      platform = if platform_exception == :ceph
+        target_platform(platform_exception: :ceph)
+      else
+        target_platform
+      end
+
+      {}.tap do |ret|
+        NodeObject.all.map(&:architecture).each do |arch|
+          repos = ::Crowbar::Repository.registry[platform][arch].select do |_name, values|
+            values["features"] == ["os"] && values["required"] == "mandatory"
+          end.keys
+
+          repos.each do |repo|
+            ret[repo] = {}
+            repository = ::Crowbar::Repository.where(
+              platform: platform,
+              arch: arch,
+              repo: repo
+            ).first
+            ret[repo]["available"] = repository.available?
+            next if repository.active?
+
+            ProvisionerService.new(Rails.logger).enable_repository(
+              platform: platform,
+              arch: arch,
+              repo: repo
+            )
+            ret[repo]["available"] = repository.active?
+          end
+        end
+      end
+    end
+
+    def target_platform(options = {})
+      # FIXME: get rid of platform_exception when SES upgraded to SP2
+      platform_exception = options.fetch(:platform_exception, nil)
+
+      case Api::Crowbar.new.version
+      when "4.0"
+        if platform_exception == :ceph
+          "suse-12.1"
```

Comment:
You should use the method from https://github.com/crowbar/crowbar-core/blob/master/crowbar_framework/lib/crowbar/product.rb instead of hard-coding.